src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -198,11 +198,38 @@ public int BytesTransferred
 
         public event EventHandler<SocketAsyncEventArgs>? Completed;
 
+        private void OnCompletedInternal()
+        {
+            if (SocketsTelemetry.Log.IsEnabled())
+            {
+                AfterConnectAcceptTelemetry();
+            }
+
+            OnCompleted(this);
+        }
+
         protected virtual void OnCompleted(SocketAsyncEventArgs e)
         {
             Completed?.Invoke(e._currentSocket, e);
         }
 
+        private void AfterConnectAcceptTelemetry()
+        {
```

Comment:
I would personally prefer the pattern of checking `IsEnabled` only in `*Telemetry.cs` code, but checking it before doing anything Telemetry-related at all call sites is the pattern followed everywhere else.

Reply:
> I would personally prefer the pattern of checking `IsEnabled` only in `*Telemetry.cs` code

Totally agree. My personal feeling is that Socket code became now super-bloated with telemetry calls, which makes it harder to maintain. I wonder if we can propose to change the pattern we use for Telemetry. When we do not do string interpolation, I see no other argument against moving the `if (SocketsTelemetry.Log.IsEnabled())` checks there, except consistency with the interpolating logs which are in minority.

Reply:
> I see no other argument against moving the if (SocketsTelemetry.Log.IsEnabled()) checks there

It could potentially impact perf if it's a hot path and the method doesn't get inlined.

Reply:
Can't this be fixed by careful combination of `MethodImplOptions.AggressiveInlining` for hot  `MethodImplOptions.NoInlining` for cold bits?